pkg/crc/preflight/preflight_darwin.go

```diff
@@ -107,24 +107,24 @@ var traySetupChecks = [...]Check{
 }
 
 func getAllPreflightChecks() []Check {
-	return getPreflightChecks(true, network.DefaultMode)
+	return getPreflightChecks(true, true, network.DefaultMode)
 }
 
-func getPreflightChecks(experimentalFeatures bool, mode network.Mode) []Check {
+func getPreflightChecks(_ bool, trayAutostart bool, mode network.Mode) []Check {
 	checks := []Check{}
 
 	checks = append(checks, nonWinPreflightChecks[:]...)
 	checks = append(checks, genericPreflightChecks[:]...)
 	checks = append(checks, hyperkitPreflightChecks(mode)...)
 	checks = append(checks, dnsPreflightChecks[:]...)
 
+	if trayAutostart {
+		checks = append(checks, traySetupChecks[:]...)
```

Comment:
Can you separate the daemon from the tray so that the daemon is always installed now ? We can do it separately (PR or commit) if you prefer also.

Reply:
Sure, will add another commit for it.

Reply:
If "daemon is always installed" means "daemon is always running" regardless of the tray being used or not, this is only needed for vsock, not with the default networking.

Reply:
Separated the daemon checks, and these are ran if network mode is vsock or if tray is set to autostart